rapids-cmake/cpm/versions.json

```diff
@@ -18,7 +18,14 @@
       "git_tag" : "v2.2.0",
       "proprietary_binary" : {
         "x86_64-linux" : "https://developer.download.nvidia.com/compute/nvcomp/${version}/local_installers/nvcomp_install_CUDA_11.x.tgz"
-      }
+      },
+      "patches" : [
+        {
+          "file" : "nvcomp/bad_cudart_dep.diff",
+          "issue" : "leaking cudart static dep to consumers",
+          "fixed_in" : "2.3.3"
```

Comment:
I think the previous patch condition was `==2.3.2` but this will patch `<2.3.3`. Is that desired? Will the patch apply to earlier versions successfully? Is that support important for users (if any) relying on the last open-source version of 2.2?

Reply:
Good questions. The patch version check is computed based on the `version` entry value. So for nvcomp the `patches` entries won't be applied to the open source fallback since the current `version` value matches the `fixed_in` value.

If we rolled back the `version` value ( say to 2.3.1 ) we will have the patch step execute and fail, but the entire CMake configuration will be considered valid:

```cmake
-- CPM: adding package nvcomp@2.3.1 (v2.2.0)
Release build.
-- Finding CUB
Build nvCOMP version 2.2.0
-- rapids-cmake [nvcomp]: failed to apply diff bad_cudart_dep.diffrapids-cmake [nvcomp]: git diff output: error: lib/cmake/nvcomp/nvcomp-targets.cmake: No such file or directory
-- Configuring done
-- Generating done
-- Build files have been written to: /home/rmaynard/Work/temp/build
```

Reply:
If we do not need to support 2.3.1 then this is fine.

Reply:
I think we should "live at head" and drop the nvcomp and thrust 1.17 patches. Users that are at head won't need these patches, and people that use an override for older versions can explicit specify the patches themselves.

Thoughts?

Reply:
I'm fine with that!

Reply:
rewrote history and dropped the unneeded patches.